### DISCUSSION TITLE
Recommended way for docker-compose setup

### DISCUSSION THREAD

**Discussion by simibac** [decentralized-identity/veramo#499] [General]
I am trying to setup completely local environment with multiple cloud agents (issuer, verifier, holder). I am using [this repo](https://github.com/uport-project/veramo-agent-deploy) to build the containers. However, running locally does not seem to work well with DIDs since the url contains a `:` in the url. This leads to problems when resolving the DIDs since this is a significant delimiter. 
The other approach is to use the Veramo CLI and run `veramo server`. However, the generated ngrok url changes on every startup and adds additional complexity during configuration.

What is the recommended way for setting up a completely local dev environment?

**Comment by mirceanis** [comment]
Looks like [veramo-agent-deploy](https://github.com/uport-project/veramo-agent-deploy) is a few versions behind.
Veramo 1.2.0 should support URLs with the `:` character ( see #383 #384).

I'm not sure if this will be enough for local setup. `did:web` requires fetching the did.json file over `https` which, I think would need to be setup outside of veramo.
